src/ProducerConfig.cc

```diff
@@ -153,6 +162,25 @@ ProducerConfig::ProducerConfig(const Napi::Object& producerConfig) : topic("") {
                                                  value.Utf8Value().c_str());
     }
   }
+
+  if ((producerConfig.Has(CFG_PUBLIC_KEY_PATH) && producerConfig.Has(CFG_PRIVATE_KEY_PATH)) &&
+      (producerConfig.Get(CFG_PUBLIC_KEY_PATH).IsString() &&
+       producerConfig.Get(CFG_PRIVATE_KEY_PATH).IsString())) {
+    std::string publicKeyPath = producerConfig.Get(CFG_PUBLIC_KEY_PATH).ToString().Utf8Value();
+    std::string privateKeyPath = producerConfig.Get(CFG_PRIVATE_KEY_PATH).ToString().Utf8Value();
+    pulsar_producer_configuration_set_default_crypto_key_reader(this->cProducerConfig, publicKeyPath.c_str(),
+                                                                privateKeyPath.c_str());
+    if (producerConfig.Has(CFG_ENCRYPTION_KEY) && producerConfig.Get(CFG_ENCRYPTION_KEY).IsString()) {
+      std::string encryptionKey = producerConfig.Get(CFG_ENCRYPTION_KEY).ToString().Utf8Value();
+      pulsar_producer_configuration_set_encryption_key(this->cProducerConfig, encryptionKey.c_str());
+    }
+    if (producerConfig.Has(CFG_CRYPTO_FAILURE_ACTION) &&
+        producerConfig.Get(CFG_CRYPTO_FAILURE_ACTION).IsString()) {
+      std::string cryptoFailureAction = producerConfig.Get(CFG_CRYPTO_FAILURE_ACTION).ToString().Utf8Value();
+      pulsar_producer_configuration_set_crypto_failure_action(
+          this->cProducerConfig, PRODUCER_CRYPTO_FAILURE_ACTION.at(cryptoFailureAction));
```

Comment:
We should check if `cryptoFailureAction` is contained in `PRODUCER_CRYPTO_FAILURE_ACTION`.